crates/indexer/src/accountsdb/accounts/candy_machine.rs

```diff
@@ -0,0 +1,302 @@
+use futures_util::{future::join_all, Future};
+use indexer_core::{
+    db::{
+        custom_types::{
+            EndSettingType as DbEndSettingType, WhitelistMintMode as DbWhitelistMintMode,
+        },
+        insert_into,
+        models::{
+            CMCollectionPDA, CMConfigLine, CMCreator, CMEndSetting, CMGateKeeperConfig,
+            CMHiddenSetting, CMWhitelistMintSetting, CandyMachine as DbCandyMachine,
+            CandyMachineData as CMData,
+        },
+        tables::{
+            candy_machine_collection_pdas, candy_machine_config_lines, candy_machine_creators,
+            candy_machine_datas, candy_machine_end_settings, candy_machine_gate_keeper_configs,
+            candy_machine_hidden_settings, candy_machine_whitelist_mint_settings, candy_machines,
+        },
+    },
+    prelude::*,
+};
+use mpl_candy_machine::{
+    CandyMachine, CandyMachineData, CollectionPDA, ConfigLine, Creator, EndSettingType,
+    EndSettings, GatekeeperConfig, HiddenSettings, WhitelistMintMode, WhitelistMintSettings,
+};
+
+use super::Client;
+use crate::prelude::*;
+
+pub(crate) async fn process(
+    client: &Client,
+    key: Pubkey,
+    candy_machine: CandyMachine,
+) -> Result<()> {
+    let cm = DbCandyMachine {
+        address: Owned(bs58::encode(key).into_string()),
+        authority: Owned(bs58::encode(candy_machine.authority).into_string()),
+        wallet: Owned(bs58::encode(candy_machine.wallet).into_string()),
+        token_mint: candy_machine
+            .token_mint
+            .map(|t| Owned(bs58::encode(t).into_string())),
+        items_redeemed: candy_machine.items_redeemed.try_into()?,
+    };
+
+    client
+        .db()
+        .run(move |db| {
+            insert_into(candy_machines::table)
+                .values(&cm)
+                .on_conflict(candy_machines::address)
+                .do_update()
+                .set(&cm)
+                .execute(db)
+        })
+        .await
+        .context("failed to insert candy machine")?;
+
+    let mut futures: Vec<std::pin::Pin<Box<dyn Future<Output = Result<()>> + Send>>> = Vec::new();
```

Comment:
you can use a `vec![...]` literal rather than `Vec::new()` followed by `.push(...)`